packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -289,21 +311,29 @@ function objectToMapTree(data: InsertableContent, schema: TreeNodeSchema): MapTr
 	const fields = new Map<FieldKey, MapTree[]>();
 
 	// Filter keys to only those that are strings - our trees do not support symbol or numeric property keys
-	const keys = Reflect.ownKeys(data).filter((key) => typeof key === "string") as FieldKey[];
-
-	for (const viewKey of keys) {
-		const fieldValue = (data as Record<FieldKey, InsertableContent>)[viewKey];
-
-		// Omit undefined record entries - an entry with an undefined key is equivalent to no entry
-		if (fieldValue !== undefined) {
-			const fieldSchema = getObjectFieldSchema(schema, viewKey);
-			const mappedChildTree = nodeDataToMapTree(fieldValue, fieldSchema.allowedTypeSet);
-			const flexKey: FieldKey = brand(getStoredKey(viewKey, fieldSchema));
-
-			// Note: SchemaFactory validates this at schema creation time, with a user-friendly error.
-			// So we don't expect to hit this, and if we do it is likely an internal bug.
-			assert(!fields.has(flexKey), 0x925 /* Keys must not be duplicated */);
-			fields.set(flexKey, [mappedChildTree]);
+	const keys = Reflect.ownKeys(data).filter((key): key is string => typeof key === "string");
+
+	// Loop through field keys without data, and assign value from its default provider.
+	for (const [key, fieldSchema] of Object.entries(
+		schema.info as Record<string, ImplicitFieldSchema>,
+	)) {
+		if (keys.includes(key)) {
```

Comment:
This function is a lot more readable now IMO, thank you! 

One additional suggestion - get rid of the `keys` array. All that `keys` is doing is filtering to the string keys of data, but the loop here is already filtering - it's only checking against string keys anyway (because `Object.entries` only considers string-keyed properties, and even if it did include numbers/symbols, we know that `schema.info` only has string properties too!).

Here, you can just do:

```ts
const value = (data as Record<string, InsertableContent | undefined>)[key];
if (value !== undefined) {
	setFieldValue(/*...*/)
} else {
	// default provider case
}
```

This is straightforward to read now. "For each key in my schema, does my inserted content have this key? If yes, use its value. If no, do the default."

Reply:
I've tried to remove the keys array, but unfortunately I run into an issue with an edge case, where if the `key` is named the same as a built-in javascript function, the `data` object returns the function, and not undefined.

I ran into a failing test `optional shadowing builtin` in `objectNode.spec.ts`, where the key is named `toString`

Reply:
Got it. The "own" in "ownKeys" is the important part. `toString` is a key of `data`, but it is not an "own" key. So you could still do the same thing but also check that `Object.hasOwnProperty.call(x, key)` - that will exclude things like "toString". 

You can choose the approach you prefer though - this isn't a super hot path and we don't need to micro-optimize removing unnecessary arrays and array searches right now.

Reply:
Added check for `Object.hasOwnProperty.call(x, key)`